core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -655,6 +655,7 @@ class UnifiedLog(@volatile var logStartOffset: Long,
   def close(): Unit = {
     debug("Closing log")
     lock synchronized {
+      flush(logEndOffset + 1)
```

Comment:
This seems to have the same problem that the log recovery point could be moved to logEndOffset + 1, which is a bit weird?

Reply:
Actually, this doesn't work. I am trying to figure out a proper solution.